packages/gatsby/src/query/file-parser.ts

```diff
@@ -179,23 +228,17 @@ async function findGraphQLTags(
         const documentLocations = new WeakMap()
 
         const extractStaticQuery = (
-          taggedTemplateExpressPath,
+          taggedTemplateExpressPath: NodePath<TaggedTemplateExpression>,
           isHook = false
-        ) => {
+        ): void => {
           const { ast: gqlAst, text, hash, isGlobal } = getGraphQLTag(
             taggedTemplateExpressPath
           )
           if (!gqlAst) return
 
           if (isGlobal) warnForGlobalTag(file)
 
-          gqlAst.definitions.forEach(def => {
-            generateQueryName({
-              def,
-              hash,
-              file,
-            })
-          })
+          gqlAst.definitions.forEach(def => generateQueryName(def, hash, file))
 
           let templateLoc
 
```

Comment:
Can you type `docInFile` below as `IGraphQLDocumentInFile`, which will show the error with the hash type.